pkg/validation/variables/variables_project.go

```diff
@@ -85,33 +85,49 @@ func validateandReplaceForProjectSource(variables map[string]string, projectSour
 		case projectSource.Git != nil:
 			gitProject := &projectSource.Git.GitLikeProjectSource
 
-			if gitProject.CheckoutFrom != nil {
-				// validate git checkout revision
-				if gitProject.CheckoutFrom.Revision, err = validateAndReplaceDataWithVariable(gitProject.CheckoutFrom.Revision, variables); err != nil {
-					checkForInvalidError(invalidKeys, err)
-				}
-
-				// // validate git checkout remote
-				if gitProject.CheckoutFrom.Remote, err = validateAndReplaceDataWithVariable(gitProject.CheckoutFrom.Remote, variables); err != nil {
-					checkForInvalidError(invalidKeys, err)
-				}
+			if err = validateAndReplaceForGitProjectSource(variables, gitProject); err != nil {
+				checkForInvalidError(invalidKeys, err)
+			}
+		}
+	}
+
+	return newInvalidKeysError(invalidKeys)
+}
+
+// validateAndReplaceForGitProjectSource validates a project git src for global variable references and replaces them with the variable value
+func validateAndReplaceForGitProjectSource(variables map[string]string, gitProject *v1alpha2.GitLikeProjectSource) error {
+
+	var err error
```

Comment:
one level nesting can be dropped here as well.

Reply:
updated for the entire pkg and updated tests for coverage